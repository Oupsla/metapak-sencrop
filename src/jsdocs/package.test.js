'use strict';

const packageTransformer = require('./package');

describe('JSDocs', () => {
  describe('Package transformer for jsdocs', () => {
    it('should work with an empty package.json', () => {
      expect(
        packageTransformer({
          metapak: {
            data: {
              files: 'yolo.js',
            },
          },
        }),
      ).toMatchSnapshot();
    });

    it('should work with an existing package.json', () => {
      expect(
        packageTransformer({
          scripts: {
            doc: 'yolo',
          },
          metapak: {
            data: {
              files: 'yolo.js',
              ignore: ['API.md'],
            },
          },
        }),
      ).toMatchSnapshot();
    });
  });
});
