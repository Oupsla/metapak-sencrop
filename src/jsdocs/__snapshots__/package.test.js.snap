// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JSDocs Package transformer for jsdocs should work with an empty package.json 1`] = `
Object {
  "devDependencies": Object {
    "jsdoc-to-markdown": "^4.0.1",
  },
  "metapak": Object {
    "data": Object {
      "files": "yolo.js",
      "ignore": Array [
        "API.md",
      ],
    },
  },
  "scripts": Object {
    "doc": "mkdir -p .readme; echo \\"# API\\" > API.md; jsdoc2md yolo.js >> API.md",
  },
}
`;

exports[`JSDocs Package transformer for jsdocs should work with an existing package.json 1`] = `
Object {
  "devDependencies": Object {
    "jsdoc-to-markdown": "^4.0.1",
  },
  "metapak": Object {
    "data": Object {
      "files": "yolo.js",
      "ignore": Array [
        "API.md",
      ],
    },
  },
  "scripts": Object {
    "doc": "mkdir -p .readme; echo \\"# API\\" > API.md; jsdoc2md yolo.js >> API.md",
  },
}
`;
